### ISSUE TITLE
MissingDeprecated: no support for module declarations

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#21164]
Parent Issue #21113

I have read check documentation: https://checkstyle.org/checks/annotation/missingdeprecated.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Note: module-info.java is only parseable after #8240, so this is reproduced on that branch.

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="MissingDeprecated"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > md/module-info.java <<'EOF'
/**
 * Javadoc without the deprecated tag.
 */
@Deprecated
module com.example.app {
    requires java.base;
}
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml md/module-info.java

Starting audit...
Audit done.
```

The identical shape on a package declaration is reported:

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > package-info.java <<'EOF'
/**
 * Javadoc without the deprecated tag.
 */
@Deprecated
package com.example;
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml package-info.java

Starting audit...
[ERROR] package-info.java:5: Must include both @java.lang.Deprecated annotation and @deprecated Javadoc tag with description. [MissingDeprecated]
Audit done.
Checkstyle ends with 1 errors.
```

**Describe what you expect in detail.**

A module declaration annotated `@Deprecated` whose Javadoc lacks a `@deprecated` tag should be reported, the same as a package declaration.

Two separate gaps block this. First, Javadoc Checks process a comment only when `JavadocUtil.isJavadocComment` returns true, and `BlockCommentPosition` has no `MODULE_DEF` branch, so the module Javadoc is not recognised as Javadoc. That part is covered by #21112. Second, the Check resolves the annotation through `AnnotationUtil.getAnnotationHolder`, which returns `ANNOTATIONS` for `PACKAGE_DEF` and `ENUM_CONSTANT_DEF` and `MODIFIERS` for everything else. `MODULE_DEF` has no `MODIFIERS` child, so the lookup returns null and `@Deprecated` is never found.

Both fixes are needed for this Check to work on module files.

Expected on the module input:

```
5: Must include both @java.lang.Deprecated annotation and @deprecated Javadoc tag with description.
```

**Comment by sahitya0xsingh** [comment]
I am on it.

**Comment by romani** [comment]
Actual fix was done in some other update .

Now we just merged update to add tests to confirm it works
